api/routes/profile_page.py

```diff
@@ -45,36 +50,38 @@ async def update_profile(userDetails: ProfileUpdate, current_user: User = Depend
             status_code=400, detail=settings.ERRORS.get("UNKNOWN ERROR"))
 
 
-@profile_route.get("/", response_model=PaginatedUsers)
-def get_all_profile(limit: int = Query(default=50, ge=1, le=100), page: int = Query(default=1, ge=1), db: Session = Depends(get_db)):
-    total_users = db.query(User).count()
-    pages = (total_users - 1) // limit + 1
-    offset = (page - 1) * limit
-    users = db.query(User).order_by(desc(User.created_at)).offset(offset).limit(limit).all()
-
-    links = {
-        "first": f"/api/v1/users/?limit={limit}&page=1",
-        "last": f"/api/v1/users/?limit={limit}&page={pages}",
-        "self": f"/api/v1/users/?limit={limit}&page={page}",
-        "next": None,
-        "prev": None,
-    }
-
-    if page < pages:
-        links["next"] = f"/api/v1/users/?limit={limit}&page={page + 1}"
-
-    if page > 1:
-        links["prev"] = f"/api/v1/users/?limit={limit}&page={page - 1}"
-
-    return PaginatedUsers(
-        users=users,
-        total=total_users,
-        page=page,
-        size=limit,
-        pages=pages,
-        links=links,
-    )
+@profile_route.get("/", response_model=Page[ProfileResponse])
+def get_all_profile(skill: str = Query(None, title="Skill", description="The skill to filter users"), 
+                    stack: str = Query(None, title="Stack", description="The stack to filter users"),
+                    active: Optional[bool] = None, p: Optional[str] = None,
+                    db: Session = Depends(get_db)):
 
+    query = db.query(User)
```

Comment:
It's a good idea to have a base query here. But I don't think the code has made provision for multiple parameters. What if I want to search for active users with Backend stack and Python skill? I think the base should be overridden in each condition. Or how ever you think it should be done.